store.go

```diff
@@ -1523,22 +1522,12 @@ func (s *store) Mount(id, mountLabel string) (string, error) {
 	if modified, err := rlstore.Modified(); modified || err != nil {
 		rlstore.Load()
 	}
-	rcstore.Lock()
-	defer rcstore.Unlock()
-	if modified, err := rcstore.Modified(); modified || err != nil {
-		rcstore.Load()
-	}
-
-	if c, err := rcstore.Get(id); c != nil && err == nil {
-		id = c.LayerID
-	}
 	return rlstore.Mount(id, mountLabel)
 }
 
 func (s *store) Unmount(id string) error {
-	rcstore, err := s.ContainerStore()
-	if err != nil {
-		return err
+	if layerID, err := s.ContainerLayerID(id); err == nil {
+		id = layerID
 	}
```

Comment:
Shouldn't an error be returned here if `ContainerLayerID` returns an error, instead of falling through and letting the layer store's `Unmount()` method eventually return an `ErrLayerUnknown` error?